spec/kinesis/consumer_efo_spec.rb

```diff
@@ -0,0 +1,221 @@
+# frozen_string_literal: true
+
+require 'rspec'
+require 'spec_helper'
+require 'kinesis/consumer'
+require 'aws-sdk-kinesis'
+
+# rubocop:disable Metrics/BlockLength
+describe Kinesis::Consumer, 'with Enhanced Fan-Out', integration: true do
+  let(:stream_arn) { 'arn:aws:kinesis:us-east-1:123456789012:stream/test-stream' }
+  let(:consumer_arn) { "#{stream_arn}/consumer/test-consumer:1234567890" }
+
+  let(:kinesis_client) do
+    client = Aws::Kinesis::Client.new(stub_responses: true)
+
+    # Stub describe_stream with all required parameters
+    client.stub_responses(:describe_stream, {
+                            stream_description: {
+                              stream_name: 'test-stream',
+                              stream_arn: stream_arn,
```

Comment:
nit: I think we're using > ruby 3.1 here, could we use the shorthand hash syntax `stream_arn:` in general for better readability?

Reply:
sure!